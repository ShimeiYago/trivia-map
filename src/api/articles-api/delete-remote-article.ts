import { BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';

export async function deleteRemoteArticle(postId: number): Promise<unknown> {
  const axiosInstance = getAxiosInstance({}, {});

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<unknown> = await axiosInstance.delete(
      `${BASE_URL}/articles/${postId}`,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}
