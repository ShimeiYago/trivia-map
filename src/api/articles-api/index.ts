import { BASE_URL } from '../constants';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import {
  mockGetArticleResponse,
  mockPostArticleResponse,
} from '../mock/articles-response';
import { Position } from 'types/position';

export async function getRemoteArticle(
  postId: string,
): Promise<GetArticleResponse> {
  const axiosInstance = getAxiosInstance({}, mockGetArticleResponse);

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<GetArticleResponse> = await axiosInstance.get(
      `${BASE_URL}/articles/${postId}`,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export async function postRemoteArticle(
  title: string,
  content: string,
  position: Position,
): Promise<PostArticleResponse> {
  const axiosInstance = getAxiosInstance({}, mockPostArticleResponse);

  const requestData: PostArticleRequest = {
    title: title,
    content: content,
    position: position,
  };

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<PostArticleResponse> = await axiosInstance.post(
      `${BASE_URL}/articles`,
      requestData,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError<ValidationError>(axiosError);
  }
}

export async function putRemoteArticle(
  postId: string,
  title: string,
  content: string,
  position: Position,
): Promise<PostArticleResponse> {
  const axiosInstance = getAxiosInstance({}, mockPostArticleResponse);

  const requestData: PostArticleRequest = {
    title: title,
    content: content,
    position: position,
  };

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<PostArticleResponse> = await axiosInstance.put(
      `${BASE_URL}/articles/${postId}`,
      requestData,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError<ValidationError>(axiosError);
  }
}

export type GetArticleResponse = {
  title: string;
  content: string;
};

export type PostArticleRequest = {
  title: string;
  content: string;
  position: Position;
};

export type PostArticleResponse = {
  postId: string;
};

type ValidationError = {
  headerErrors: string[];
  fieldErrors?: {
    title?: string[];
    content?: string[];
    position?: string[];
  };
};
