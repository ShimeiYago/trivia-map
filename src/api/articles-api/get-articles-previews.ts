import { BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockGetArticlesPreviewsResponse } from '../mock/articles-response';
import { PaginationResponse } from 'api/types/pagination-response';
import { Park } from './../../types/park';
import { getUrlParameters } from 'utils/get-url-parameters';

export async function getArticlesPreviews(
  param: GetArticlesPreviewsParam,
): Promise<GetArticlesPreviewsResponse> {
  const axiosInstance = getAxiosInstance({}, mockGetArticlesPreviewsResponse);

  const urlParams = getUrlParameters(param);
  const url = `${BASE_URL}/articles/public/previews${urlParams}`;

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<GetArticlesPreviewsResponse> =
      await axiosInstance.get(url);
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type GetArticlesPreviewsResponseEachItem = {
  postId: number;
  title: string;
  image: string | null;
  category: number;
};

export type GetArticlesPreviewsResponse =
  PaginationResponse<GetArticlesPreviewsResponseEachItem>;

export type GetArticlesPreviewsParam = {
  page?: number;
  marker?: number;
  user?: number;
  category?: number;
  park?: Park;
  keywords?: string[];
};
