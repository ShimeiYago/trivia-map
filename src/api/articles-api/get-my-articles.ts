import { API_TIMEOUT, BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockGetMyArticlesResponse } from '../mock/articles-response';
import { PaginationResponse } from 'api/types/pagination-response';
import { Park } from 'types/park';
import { getUrlParameters } from 'utils/get-url-parameters';

export async function getMyArticles(param: GetMyArticlesParam): Promise<GetMyArticlesResponse> {
  const axiosInstance = getAxiosInstance({ timeout: API_TIMEOUT.short }, mockGetMyArticlesResponse);

  const urlParams = getUrlParameters(param);
  const url = `${BASE_URL}/articles/mine${urlParams}`;

  try {
    const res: AxiosResponse<GetMyArticlesResponse> = await axiosInstance.get(url);
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type GetMyArticlesResponseEachItem = {
  postId: number;
  title: string;
  category: number;
  isDraft: boolean;
  image: string | null;
  numberOfGoods: number;
};

export type GetMyArticlesParam = {
  page?: number;
  category?: number;
  park?: Park;
  isDraft?: 'true' | 'false';
};

export type GetMyArticlesResponse = PaginationResponse<GetMyArticlesResponseEachItem>;
