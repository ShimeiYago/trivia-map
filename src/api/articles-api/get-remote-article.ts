import { BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockGetArticleResponse } from '../mock/articles-response';
import { Position } from 'types/position';

export async function getRemoteArticle(
  postId: string,
): Promise<GetArticleResponse> {
  const axiosInstance = getAxiosInstance({}, mockGetArticleResponse);

  try {
    // TODO: Set reasonable timeout
    const res: AxiosResponse<GetArticleResponse> = await axiosInstance.get(
      `${BASE_URL}/articles/${postId}`,
    );
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type GetArticleResponse = {
  title: string;
  content: string;
  position: Position;
  imageDataUrl: string | null;
};
