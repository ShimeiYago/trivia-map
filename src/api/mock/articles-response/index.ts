import { GetArticlesPreviewsResponse } from 'api/articles-api/get-articles-previews';
import { GetMyArticlesResponse } from 'api/articles-api/get-my-articles';
import { GetArticleResponse } from 'api/articles-api/get-remote-article';
import { PostArticleResponse } from 'api/articles-api/post-remote-article';

export const mockGetArticleResponse: GetArticleResponse = {
  postId: 1,
  title: 'ノーチラス号',
  description: 'ネモ船長が最大の発明だと自負する潜水艇。',
  marker: {
    markerId: 1,
    lat: -111.33,
    lng: 149.243,
    park: 'S',
    numberOfPublicArticles: { total: 4, eachCategory: [1, 2, 1] },
    areaNames: ['シー', 'メディテレーニアンハーバー', 'ポルトパラディーゾ'],
  },
  image: 'https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg',
  isDraft: false,
  author: {
    userId: 1,
    nickname: 'Axel',
    icon: 'https://disneyparkstory.com/wp-content/uploads/2017/09/apple-touch-icon.png',
  },
  createdAt: '2022/06/16 21:42',
  updatedAt: '2022/06/16 21:42',
  category: 1,
  numberOfLikes: 1,
};

export const mockPostArticleResponse: PostArticleResponse = {
  postId: 1,
  title: 'ノーチラス号',
  description: 'ネモ船長が最大の発明だと自負する潜水艇。',
  marker: 100,
  image: null,
  isDraft: false,
  author: 0,
  category: 1,
  createdAt: '2022/06/16 21:42',
  updatedAt: '2022/06/16 21:42',
};

export const mockGetArticlesPreviewsResponse: GetArticlesPreviewsResponse = {
  nextUrl: null,
  previousUrl: null,
  totalRecords: 2,
  totalPages: 2,
  currentPage: 1,
  startIndex: 1,
  endIndex: 1,
  results: [
    {
      postId: 1,
      title: 'ノーチラス号のエンジン',
      image: 'https://disneyparkstory.com/wp-content/uploads/2017/02/CIMG9800.jpg',
      category: 1,
      createdAt: '2022/06/16 21:42',
      numberOfLikes: 1,
    },
    {
      postId: 2,
      title: 'ノーチラス号の名前の由来',
      image: null,
      category: 1,
      createdAt: '2022/06/16 21:42',
      numberOfLikes: 1,
    },
  ],
};

export const mockGetMyArticlesResponse: GetMyArticlesResponse = {
  nextUrl: null,
  previousUrl: null,
  totalRecords: 2,
  totalPages: 2,
  startIndex: 1,
  endIndex: 1,
  currentPage: 1,
  results: [
    {
      postId: 1,
      title: 'ノーチラス号のエンジン',
      isDraft: false,
      category: 1,
    },
    {
      postId: 2,
      title: 'ノーチラス号の名前の由来',
      isDraft: true,
      category: 1,
    },
  ],
};
