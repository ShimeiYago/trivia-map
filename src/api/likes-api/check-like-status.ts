import { API_TIMEOUT, BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockCheckLikeStatusResponse } from '../mock/likes-response';

export async function checkLikeStatus(postId: number): Promise<CheckLikeStatusResponse> {
  const axiosInstance = getAxiosInstance(
    { timeout: API_TIMEOUT.short },
    mockCheckLikeStatusResponse,
  );

  const url = `${BASE_URL}/likes/check/${postId}`;

  try {
    const res: AxiosResponse<CheckLikeStatusResponse> = await axiosInstance.get(url);
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type CheckLikeStatusResponse = {
  haveLiked: boolean;
};
