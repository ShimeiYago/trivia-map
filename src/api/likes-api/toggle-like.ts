import { API_TIMEOUT, BASE_URL } from 'constant';
import { AxiosError, AxiosResponse } from 'axios';
import { handleAxiosError } from '../utils/handle-axios-error';
import { getAxiosInstance } from 'api/utils/get-axios-instance';
import { mockToggleLikeResponse } from '../mock/likes-response';

export async function toggleLike(postId: number): Promise<ToggleLikeResponse> {
  const axiosInstance = getAxiosInstance({ timeout: API_TIMEOUT.long }, mockToggleLikeResponse);

  const url = `${BASE_URL}/likes/toggle/${postId}`;

  try {
    const res: AxiosResponse<ToggleLikeResponse> = await axiosInstance.post(url);
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError;
    throw handleAxiosError(axiosError);
  }
}

export type ToggleLikeResponse = {
  haveLiked: boolean;
};
