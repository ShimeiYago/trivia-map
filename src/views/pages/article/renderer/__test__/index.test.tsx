import { GetArticleResponse } from 'api/articles-api/get-remote-article';
import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, Props, State } from '..';
import * as GetRemoteArticleModule from 'api/articles-api/get-remote-article';
import * as CheckLikeStatusModule from 'api/likes-api/check-like-status';
import * as ToggleLikeModule from 'api/likes-api/toggle-like';
import { mockGetArticleResponse } from 'api/mock/articles-response';

let wrapper: ShallowWrapper<Props, State, Renderer>;

let getRemoteArticleSpy: jest.SpyInstance;
let checkLikeStatusSpy: jest.SpyInstance;
let toggleLikeSpy: jest.SpyInstance;

const basicProps: Props = {
  postId: 1,
  focusingPark: 'S',
  initialize: jest.fn(),
  throwError: jest.fn(),
  refreshUser: jest.fn(),
  toggleAuthFormModal: jest.fn(),
  updateInitMapFocus: jest.fn(),
  updateFocusingPark: jest.fn(),
  initializeFetchingState: jest.fn(),
};

const testUser = {
  nickname: 'Axel',
  userId: 1,
  email: 'xxx',
  icon: null,
  isSocialAccount: false,
};

const article: GetArticleResponse = {
  postId: 1,
  title: 'title',
  description: 'description',
  marker: {
    markerId: 1,
    lat: 0,
    lng: 0,
    park: 'S',
    numberOfPublicArticles: { total: 4, eachCategory: [1, 2, 1] },
    areaNames: ['シー'],
  },
  image: null,
  author: {
    userId: 1,
    nickname: 'Axel',
    icon: null,
  },
  createdAt: '2022/4/1',
  updatedAt: '2022/5/1',
  isDraft: false,
  category: 1,
  numberOfLikes: 1,
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<Renderer {...basicProps} />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('with article', () => {
    wrapper.setState({ article: article, loadingArticleState: 'success' });
    expect(wrapper).toMatchSnapshot();
  });

  it('loading', () => {
    wrapper.setState({ loadingArticleState: 'loading' });
    expect(wrapper).toMatchSnapshot();
  });

  it('with image', () => {
    wrapper.setState({
      article: { ...article, image: 'image.jpg' },
      loadingArticleState: 'success',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('draft', () => {
    wrapper.setState({
      article: { ...article, isDraft: true },
      loadingArticleState: 'success',
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('author view', () => {
    wrapper.setProps({
      user: testUser,
    });
    wrapper.setState({ article: article, loadingArticleState: 'success' });
    expect(wrapper).toMatchSnapshot();
  });

  it('have liked', () => {
    wrapper.setState({
      haveLiked: true,
      loadingArticleState: 'success',
      article: article,
      numberOfLikes: 0,
    });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('componentDidUpdate', () => {
  beforeEach(() => {
    getRemoteArticleSpy = jest.spyOn(GetRemoteArticleModule, 'getRemoteArticle');
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should call fetchArticle if postId is updated', async () => {
    getRemoteArticleSpy.mockResolvedValue(mockGetArticleResponse);

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    instance['componentDidUpdate']({
      ...basicProps,
      postId: 2,
    });

    expect(getRemoteArticleSpy).toBeCalled();
  });
});

describe('fetchArticle', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    getRemoteArticleSpy = jest.spyOn(GetRemoteArticleModule, 'getRemoteArticle');
  });

  it('should set loadingArticleState success when api succeed', async () => {
    getRemoteArticleSpy.mockResolvedValue({});

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    await instance['fetchArticle']();

    expect(instance.state.loadingArticleState).toBe('success');
  });

  it('should set redirectNotFound when api fail with 404', async () => {
    getRemoteArticleSpy.mockRejectedValue({ status: 404 });

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    await instance['fetchArticle']();

    expect(instance.props.throwError).toBeCalled();
  });

  it('should set loadingstate error when api fail', async () => {
    getRemoteArticleSpy.mockRejectedValue(new Error());

    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    await instance['fetchArticle']();

    expect(instance.props.throwError).toBeCalled();
  });

  it('should call initializeFetchingState if focusingPark is changed', async () => {
    getRemoteArticleSpy.mockResolvedValue(mockGetArticleResponse);

    wrapper = shallow(<Renderer {...basicProps} focusingPark="L" />);
    const instance = wrapper.instance();

    await instance['fetchArticle']();

    expect(instance.props.initializeFetchingState).toBeCalled();
  });
});

describe('checkLikeStatus', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    checkLikeStatusSpy = jest.spyOn(CheckLikeStatusModule, 'checkLikeStatus');
  });

  it('should set loadingLikeState success when api succeed', async () => {
    checkLikeStatusSpy.mockResolvedValue({
      haveLiked: true,
    });

    wrapper = shallow(<Renderer {...basicProps} user={testUser} />);
    const instance = wrapper.instance();

    await instance['checkLikeStatus']();

    expect(instance.state.loadingLikeState).toBe('success');
  });

  it('should set loadingstate error when api fail', async () => {
    checkLikeStatusSpy.mockRejectedValue(new Error());

    wrapper = shallow(<Renderer {...basicProps} user={testUser} />);
    const instance = wrapper.instance();

    await instance['checkLikeStatus']();

    expect(instance.props.throwError).toBeCalled();
  });
});

describe('handleClickLikeButton', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    toggleLikeSpy = jest.spyOn(ToggleLikeModule, 'toggleLike');
  });

  it('should set loadingLikeState success when api succeed', async () => {
    toggleLikeSpy.mockResolvedValue({
      haveLiked: true,
      numberOfLikes: 1,
    });

    wrapper = shallow(<Renderer {...basicProps} user={testUser} />);
    const instance = wrapper.instance();

    await instance['handleClickLikeButton']();

    expect(instance.state.loadingLikeState).toBe('success');
  });

  it('should set loadingstate error when api fail', async () => {
    toggleLikeSpy.mockRejectedValue(new Error());

    wrapper = shallow(<Renderer {...basicProps} user={testUser} />);
    const instance = wrapper.instance();

    await instance['handleClickLikeButton']();

    expect(instance.props.throwError).toBeCalled();
  });

  it('should call toggleAuthFormModal if user is undefined', async () => {
    wrapper = shallow(<Renderer {...basicProps} />);
    const instance = wrapper.instance();

    await instance['handleClickLikeButton']();

    expect(instance.props.toggleAuthFormModal).toBeCalled();
  });
});
