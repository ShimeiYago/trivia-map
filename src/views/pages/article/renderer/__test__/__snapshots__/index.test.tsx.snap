// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests author view 1`] = `
<ArticleWrapper
  showSidebar={true}
>
  <CommonHelmet
    description="description"
    ogType="article"
    title="title - トリビアマップ"
  />
  <ForwardRef(Stack)
    spacing={2}
  >
    <ForwardRef(Link)
      component="div"
    >
      <IconAndText
        align="right"
        iconComponent={<Memo />}
        iconPosition="left"
        link="/map/edit/1"
        text="編集"
      />
    </ForwardRef(Link)>
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      justifyContent="flex-start"
      spacing={1}
    >
      <NonStyleLink
        to="/users/1"
      >
        <ForwardRef(Avatar)
          src="no-icon.jpg"
          sx={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </NonStyleLink>
      <NonStyleLink
        to="/users/1"
      >
        <ForwardRef(Typography)
          color="gray"
          sx={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          Axel
        </ForwardRef(Typography)>
      </NonStyleLink>
    </ForwardRef(Stack)>
    <ForwardRef(Typography)
      align="center"
      component="h2"
      variant="h4"
    >
      title
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="gray"
      component="div"
      sx={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      <IconAndText
        align="left"
        iconComponent={<Memo />}
        iconPosition="left"
        link="/categories/1"
        text="隠れミッキー"
      />
    </ForwardRef(Typography)>
    <ForwardRef(Box)
      sx={
        Object {
          "color": "gray",
          "textAlign": "right",
        }
      }
    >
      <ForwardRef(Typography)>
        投稿日 2022/4/1
      </ForwardRef(Typography)>
      <ForwardRef(Typography)>
        更新日 2022/5/1
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
    <ForwardRef(Divider) />
    <ForwardRef(Typography)>
      description
    </ForwardRef(Typography)>
    <ForwardRef(Divider) />
    <IconAndText
      component="h3"
      iconComponent={<Memo />}
      iconPosition="left"
      text="地図"
      variant="h5"
    />
    <AreaNames
      areaNames={
        Array [
          "シー",
        ]
      }
      variant="body2"
    />
    <TriviaMap
      additinalMarkers={
        Array [
          Object {
            "areaNames": Array [
              "シー",
            ],
            "lat": 0,
            "lng": 0,
            "markerId": 1,
            "numberOfPublicArticles": 1,
            "park": "S",
          },
        ]
      }
      disabled={true}
      doNotShowPostMarkers={true}
      height={300}
      initCenter={
        Object {
          "areaNames": Array [
            "シー",
          ],
          "lat": 0,
          "lng": 0,
          "markerId": 1,
          "numberOfPublicArticles": 1,
          "park": "S",
        }
      }
      initZoom={3}
      park="S"
    />
  </ForwardRef(Stack)>
</ArticleWrapper>
`;

exports[`Shallow Snapshot Tests basic 1`] = `
<ArticleWrapper
  showSidebar={true}
>
  <CenterSpinner />
</ArticleWrapper>
`;

exports[`Shallow Snapshot Tests draft 1`] = `
<ArticleWrapper
  showSidebar={true}
>
  <CommonHelmet
    description="description"
    ogType="article"
    title="title - トリビアマップ"
  />
  <ForwardRef(Stack)
    spacing={2}
  >
    <ForwardRef(Alert)
      severity="info"
    >
      この記事は下書きです。あなただけが閲覧できます。
    </ForwardRef(Alert)>
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      justifyContent="flex-start"
      spacing={1}
    >
      <NonStyleLink
        to="/users/1"
      >
        <ForwardRef(Avatar)
          src="no-icon.jpg"
          sx={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </NonStyleLink>
      <NonStyleLink
        to="/users/1"
      >
        <ForwardRef(Typography)
          color="gray"
          sx={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          Axel
        </ForwardRef(Typography)>
      </NonStyleLink>
    </ForwardRef(Stack)>
    <ForwardRef(Typography)
      align="center"
      component="h2"
      variant="h4"
    >
      title
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="gray"
      component="div"
      sx={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      <IconAndText
        align="left"
        iconComponent={<Memo />}
        iconPosition="left"
        link="/categories/1"
        text="隠れミッキー"
      />
    </ForwardRef(Typography)>
    <ForwardRef(Box)
      sx={
        Object {
          "color": "gray",
          "textAlign": "right",
        }
      }
    >
      <ForwardRef(Typography)>
        投稿日 2022/4/1
      </ForwardRef(Typography)>
      <ForwardRef(Typography)>
        更新日 2022/5/1
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
    <ForwardRef(Divider) />
    <ForwardRef(Typography)>
      description
    </ForwardRef(Typography)>
    <ForwardRef(Divider) />
    <IconAndText
      component="h3"
      iconComponent={<Memo />}
      iconPosition="left"
      text="地図"
      variant="h5"
    />
    <AreaNames
      areaNames={
        Array [
          "シー",
        ]
      }
      variant="body2"
    />
    <TriviaMap
      additinalMarkers={
        Array [
          Object {
            "areaNames": Array [
              "シー",
            ],
            "lat": 0,
            "lng": 0,
            "markerId": 1,
            "numberOfPublicArticles": 1,
            "park": "S",
          },
        ]
      }
      disabled={true}
      doNotShowPostMarkers={true}
      height={300}
      initCenter={
        Object {
          "areaNames": Array [
            "シー",
          ],
          "lat": 0,
          "lng": 0,
          "markerId": 1,
          "numberOfPublicArticles": 1,
          "park": "S",
        }
      }
      initZoom={3}
      park="S"
    />
  </ForwardRef(Stack)>
</ArticleWrapper>
`;

exports[`Shallow Snapshot Tests loading 1`] = `
<ArticleWrapper
  showSidebar={true}
>
  <CenterSpinner />
</ArticleWrapper>
`;

exports[`Shallow Snapshot Tests with article 1`] = `
<ArticleWrapper
  showSidebar={true}
>
  <CommonHelmet
    description="description"
    ogType="article"
    title="title - トリビアマップ"
  />
  <ForwardRef(Stack)
    spacing={2}
  >
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      justifyContent="flex-start"
      spacing={1}
    >
      <NonStyleLink
        to="/users/1"
      >
        <ForwardRef(Avatar)
          src="no-icon.jpg"
          sx={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </NonStyleLink>
      <NonStyleLink
        to="/users/1"
      >
        <ForwardRef(Typography)
          color="gray"
          sx={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          Axel
        </ForwardRef(Typography)>
      </NonStyleLink>
    </ForwardRef(Stack)>
    <ForwardRef(Typography)
      align="center"
      component="h2"
      variant="h4"
    >
      title
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="gray"
      component="div"
      sx={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      <IconAndText
        align="left"
        iconComponent={<Memo />}
        iconPosition="left"
        link="/categories/1"
        text="隠れミッキー"
      />
    </ForwardRef(Typography)>
    <ForwardRef(Box)
      sx={
        Object {
          "color": "gray",
          "textAlign": "right",
        }
      }
    >
      <ForwardRef(Typography)>
        投稿日 2022/4/1
      </ForwardRef(Typography)>
      <ForwardRef(Typography)>
        更新日 2022/5/1
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
    <ForwardRef(Divider) />
    <ForwardRef(Typography)>
      description
    </ForwardRef(Typography)>
    <ForwardRef(Divider) />
    <IconAndText
      component="h3"
      iconComponent={<Memo />}
      iconPosition="left"
      text="地図"
      variant="h5"
    />
    <AreaNames
      areaNames={
        Array [
          "シー",
        ]
      }
      variant="body2"
    />
    <TriviaMap
      additinalMarkers={
        Array [
          Object {
            "areaNames": Array [
              "シー",
            ],
            "lat": 0,
            "lng": 0,
            "markerId": 1,
            "numberOfPublicArticles": 1,
            "park": "S",
          },
        ]
      }
      disabled={true}
      doNotShowPostMarkers={true}
      height={300}
      initCenter={
        Object {
          "areaNames": Array [
            "シー",
          ],
          "lat": 0,
          "lng": 0,
          "markerId": 1,
          "numberOfPublicArticles": 1,
          "park": "S",
        }
      }
      initZoom={3}
      park="S"
    />
  </ForwardRef(Stack)>
</ArticleWrapper>
`;

exports[`Shallow Snapshot Tests with image 1`] = `
<ArticleWrapper
  showSidebar={true}
>
  <CommonHelmet
    description="description"
    imageUrl="image.jpg"
    ogType="article"
    title="title - トリビアマップ"
  />
  <ForwardRef(Stack)
    spacing={2}
  >
    <ForwardRef(Stack)
      alignItems="center"
      direction="row"
      justifyContent="flex-start"
      spacing={1}
    >
      <NonStyleLink
        to="/users/1"
      >
        <ForwardRef(Avatar)
          src="no-icon.jpg"
          sx={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </NonStyleLink>
      <NonStyleLink
        to="/users/1"
      >
        <ForwardRef(Typography)
          color="gray"
          sx={
            Object {
              "textDecoration": "underline",
            }
          }
        >
          Axel
        </ForwardRef(Typography)>
      </NonStyleLink>
    </ForwardRef(Stack)>
    <ForwardRef(Typography)
      align="center"
      component="h2"
      variant="h4"
    >
      title
    </ForwardRef(Typography)>
    <ForwardRef(Typography)
      color="gray"
      component="div"
      sx={
        Object {
          "textDecoration": "underline",
        }
      }
    >
      <IconAndText
        align="left"
        iconComponent={<Memo />}
        iconPosition="left"
        link="/categories/1"
        text="隠れミッキー"
      />
    </ForwardRef(Typography)>
    <ForwardRef(Box)
      sx={
        Object {
          "color": "gray",
          "textAlign": "right",
        }
      }
    >
      <ForwardRef(Typography)>
        投稿日 2022/4/1
      </ForwardRef(Typography)>
      <ForwardRef(Typography)>
        更新日 2022/5/1
      </ForwardRef(Typography)>
    </ForwardRef(Box)>
    <ForwardRef(Divider) />
    <Image
      src="image.jpg"
      width="full"
    />
    <ForwardRef(Typography)>
      description
    </ForwardRef(Typography)>
    <ForwardRef(Divider) />
    <IconAndText
      component="h3"
      iconComponent={<Memo />}
      iconPosition="left"
      text="地図"
      variant="h5"
    />
    <AreaNames
      areaNames={
        Array [
          "シー",
        ]
      }
      variant="body2"
    />
    <TriviaMap
      additinalMarkers={
        Array [
          Object {
            "areaNames": Array [
              "シー",
            ],
            "lat": 0,
            "lng": 0,
            "markerId": 1,
            "numberOfPublicArticles": 1,
            "park": "S",
          },
        ]
      }
      disabled={true}
      doNotShowPostMarkers={true}
      height={300}
      initCenter={
        Object {
          "areaNames": Array [
            "シー",
          ],
          "lat": 0,
          "lng": 0,
          "markerId": 1,
          "numberOfPublicArticles": 1,
          "park": "S",
        }
      }
      initZoom={3}
      park="S"
    />
  </ForwardRef(Stack)>
</ArticleWrapper>
`;
